src/main/java/com/umbr3114/controllers/CommentController.java

```diff
@@ -0,0 +1,160 @@
+        commentList.count = commentModelList.size();
+        if (commentModelList.size() > 0){
+            commentList.lastId = commentModelList.get(commentModelList.size()-1)._id.toString();
+            commentList.postId = commentModelList.get(0).getPostIdString();
+        }
+        commentList.comments = commentModelList;
+        return commentList;
+
+    });
+
+    private static List<CommentModel> getAfterId(String postId, String startAfterId){
+        JacksonMongoCollection<CommentModel> mongoCollectionComments =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        JacksonMongoCollection<PostModel> mongoCollectionPost =
+                new CollectionFactory<PostModel>(ServiceLocator.getService().dbService(), PostModel.class).getCollection();
+        FindIterable<CommentModel> iterable;
+        PostModel post = mongoCollectionPost.findOne(Filters.eq("_id", postId));
+        List<CommentModel> commentModels = new ArrayList<>();
+        if(post == null){
+            return commentModels;
+        }else{
+            iterable = mongoCollectionComments.find(Filters.and(Filters.eq("post_id", postId), Filters.gt("_id", startAfterId.toString())));
+        }
+        iterable.limit(10);
+        iterable.into(commentModels);
+        return commentModels;
+    }
+
+    private static List<CommentModel> getByDate(String postId, String date){
+        JacksonMongoCollection<CommentModel> mongoCollectionComments =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        FindIterable<CommentModel> iterable;
+        List<CommentModel> commentList = new ArrayList<>();
+        iterable = mongoCollectionComments.find(Filters.and(eq("dateString", date), Filters.eq("post_id", postId))).limit(10);
+        iterable.into(commentList);
+        return commentList;
+    }
+
+    public static Route updateComments = ((request, response) -> {
+        JacksonMongoCollection<CommentModel> mongoCollection =
+                new CollectionFactory<CommentModel>(ServiceLocator.getService().dbService(), CommentModel.class).getCollection();
+        SessionManager sessionManager = new SparkSessionManager(request);
+        RequestParamHelper paramHelper = new RequestParamHelper(request);
+        CommentModel commentModel;
+        //PermissionChecker permissionChecker;
+        String commentId = paramHelper.valueOf("_id");
+        String bodyText = paramHelper.valueOf("bodyText");
+        commentModel = mongoCollection.findOne(eq("_id", new ObjectId(commentId)));
+        if(sessionManager.getCurrentUserId().equals(commentModel.authorId)){
+            commentModel.setBodyText(bodyText);
+            mongoCollection.findOneAndReplace(eq("_id", new ObjectId(commentId)), commentModel);
+            return " ok ";
+        }
+        return " not ok ";
+    });
+
```

Comment:
### `PUT /comment/update`

- server crashes with 500 error when request body is blank - should error out
  - server crashes with 500 error when request has missing `_id` parameter
  - should halt with an error instead
- still executes when body text is missing
  - should halt with an error instead
- not too big of a deal, but `_id` should probably be something like `comment_id` so that the REST api is uniform